File: bulk/javascript/path_02.js
// CWE-22: Path Traversal via res.sendFile without sanitization
const express = require('express');
const path = require('path');
const app = express();
app.get('/download', (req, res) => {
  const baseDir = path.resolve('/uploads/');
  const safeName = path.basename(req.query.name);
  const fullPath = path.join(baseDir, safeName);
  const resolvedPath = path.resolve(fullPath);

  if (!resolvedPath.startsWith(baseDir + path.sep) && resolvedPath !== baseDir) {
    return res.status(400).send('Invalid file path');
  }

  res.sendFile(resolvedPath);
});

File: bulk/go/path_02.go
// CWE-22: Path Traversal in file write
package main
import (
	"os"
	"path/filepath"
)
func saveUpload(filename string, data []byte) error {
	sanitized := filepath.Base(filename)
	fullPath := filepath.Join("/var/www/uploads", sanitized)
	return os.WriteFile(fullPath, data, 0600)
}
